### DISCUSSION TITLE
SDRTrunk Uploader

### DISCUSSION THREAD

**Discussion by robertlynch3** [chuot/rdio-scanner#300] [Show and tell]
As most know, SDRTrunk does not have a way to upload to Rdio-Scanner directly. I did not want to have to dive into the depths of SDRTrunk to add an uploader, so instead I added some middleware. [SDRTrunk-Uploader](https://github.com/robertlynch3/sdrtrunk-uploader) is a Python Flask based web server that mimics the Broadcastify Calls API. 

The architecture flow is for calls to be recorded and processed by SDRTrunk, SDRTrunk uploads the call to the SDRTrunk-Uploader middleware, SDRTrunk-Uploader then forwards the call to Rdio-Scanner.

This is really just a glorified API wrapper, but it is working.
Let me know what you think!

**Comment by atlex** [comment]
@rml596 Is it stable?
Looks like very complicated system.

**Reply by robertlynch3** [reply]
Yes this is stable. It does add an extra layer, but the mechanism is not that much additional overhead. 

Ideally SDRTrunk would natively support Rdio-Scanner, or Rdio-Scanner itself could mimic the Broadcastify API.

**Comment by hoffadg** [comment]
This caught my attention.  I've been using dirwatch SDRTrunk with rdio-uploader.  What are the advantages of using the API instead?

**Reply by robertlynch3** [reply]
You can run Rdio-Scanner and SDRTrunk on different machines. I don't know if there are any less resource utilizations. I am not sure how intensive dirwatch is.
